compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java

```diff
@@ -95,14 +95,23 @@ public SymbolEnv(BLangNode node, Scope scope) {
     }
 
     public void copyTo(SymbolEnv target) {
+        this.withoutEnclEnvCopyTo(target);
+        target.enclEnv = this;
+    }
+
+    public void exactCopyTo(SymbolEnv target) {
+        this.withoutEnclEnvCopyTo(target);
+        target.enclEnv = this.enclEnv;
+    }
+
+    private void withoutEnclEnvCopyTo(SymbolEnv target) {
```

Comment:
Can't we have an overloaded `copyTo` method that accepts an `enclEnv` instead?

Reply:
Yes, that is good, changed